src/sentry/api/endpoints/client_state.py

```diff
@@ -0,0 +1,55 @@
+# This endpoint helps managing persisted client state with a TTL for a member, organization or user
+
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.base import Endpoint
+from sentry.api.exceptions import ParameterValidationError
+from sentry.utils import json, redis
+
+STATE_CATEGORIES = {
+    "onboarding": {
+        "ttl": 30 * 24 * 60 * 60,  # the time in seconds that the state will be persisted
+        "scope": "member",  # Can be "user", "org", or "user-org"
+    }
+}
+
+
+class ClientStateEndpoint(Endpoint):
+    permission_classes = (IsAuthenticated,)
```

Comment:
@Neo-Zhixing The put endpoint should require `org:write` to use and the get endpoint should require `org:read`

Reply:
I'm using `OrganizationEndpoint` now, so this should be the default permissions now.